src/test/java/com/mongodb/hibernate/jdbc/MongoConnectionTests.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import com.mongodb.client.ClientSession;
+import java.sql.SQLException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class MongoConnectionTests {
+
+    @Mock
+    private ClientSession clientSession;
```

Comment:
The default answer `Answers.RETURNS_DEFAULTS` used by Mockito when mocking (even for `Strictness.STRICT_STUBS`, which is the default strictness) aggravates the common issues with tests that use mocks. For this reason, in the Java driver codebase we use `MongoMockito.mock` instead of `Mockito.mock` (see the documentation of `MongoMockito` for all the details).

For the same reasons `MongoMockito` was introduced in the Java driver, let's introduce it here, and use instead of default Mockito mocking.

Reply:
Nathan pointed out to `Answers.RETURNS_SMART_NULLS`, which solves the same issue as `MongoMockito`, though in a subset of situations (this subset represents the majority of situations). Also, when I tried playing with `Answers.RETURNS_SMART_NULLS`, turned out it works with `final` return classes, despite its documentation saying otherwise. That is, it works better than documented. I think, it's OK to use it instead of `MongoMockito`.